lib/beaker/dsl/helpers/host_helpers.rb

```diff
@@ -254,15 +254,29 @@ def rsync_to host, from_path, to_path, opts = {}
         # @option [String] archive_root The local directory to store the copied
         #                               file under. Defaults to
         #                               'archive/sut-files'.
+        # @option [String] archive_name The name of the archive to be copied to
+        #                               archive_root. Defaults to
+        #                               'sut-files.tgz'.
         #
         # @return [Result] Returns the result of the #scp_from operation.
-        def archive_file_from(host, from_path, opts = {}, archive_root = 'archive/sut-files')
+        def archive_file_from(host, from_path, opts = {}, archive_root = 'archive/sut-files', archive_name = 'sut-files.tgz')
+          require 'archive/tar/minitar'
           filedir = File.dirname(from_path)
           targetdir = File.join(archive_root, host.hostname, filedir)
           FileUtils.mkdir_p(targetdir)
-          scp_from(host, from_path, targetdir, opts)
+          result = scp_from(host, from_path, targetdir, opts)
+          return if result.exit_code != 0
```

Comment:
After a discussion IRL, it was discovered that `scp_from` isn't returning an error. I just updated this to raise an exception if the file doesn't exist within `archive_file_from`. Hopefully that should suffice!